extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/grpc/GrpcTracingServerInterceptor.java

```diff
@@ -116,7 +111,8 @@ protected TracingServerCallListener(final ServerCall.Listener<ReqT> delegate, fi
 
         @Override
         public void onHalfClose() {
-            try {
+            Scope scope = spanContext.makeCurrent();
+            try (scope) {
```

Comment:
```suggestion
            try (Scope scope = spanContext.makeCurrent()) {
```
Just to prevent someone to use a closed scope after the try.